library/src/updater.rs

```diff
@@ -223,11 +221,37 @@ fn update_internal(_: &UpdaterLockState) -> anyhow::Result<UpdateStatus> {
     // Saves state to disk (holds Config lock while writing).
 
     let config = copy_update_config()?;
+    // We should never try to write this state as some other writer may be
+    // racing with us, we should get a new state inside a lock if we want
+    // to write.
+    let read_only_state =
+        UpdaterState::load_or_new_on_error(&config.cache_dir, &config.release_version);
+
+    // We discard any events if we have more than 3 queued to make sure
```

Comment:
What's the significance of 3? Why not 2 or 4?

Reply:
No significance.  We could do 2 or 4 or 1 or whatever.  Originally I used 10 🤷